File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {

        int count = 0;
        int result = 0;

        for(char ch : s.toCharArray())   
        {
            if(ch == '(')
                count++;
            if(ch == ')')
                count--;
            
            result = Math.max(result, count);
        }

        return result;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function(s) {
    
    let count = 0;

    return s.split("").reduce((result, el)=>{
        if(el === '(')
            count++;
        if(el === ')')
            count--;
        
        return Math.max(result, count);
    },0);
};